src/code/views/quick-action-menu.tsx

```diff
@@ -0,0 +1,202 @@
+
+
+    const createGraphLabel = tr("~QUICK_ACTIONS.CREATE_GRAPH");
+    const createGraphClasses = "quick-action-menu-label";
+
+    const showImagesF = () => this.setState({ showImages: true });
+    const hideImagesF = () =>  this.setState({ showImages: false });
+
+    const { showGraphButton  } = this.props;
+    return(
+      <>
+        <div
+          className="quick-action-menu"
+          onMouseEnter={this.handleMouseEnter}
+          onMouseLeave={this.handleMouseLeave}
+        >
+          <div className="left-panel">
+            <ul>
+              <li
+                className={setImageClasses}
+                onClick={this.toggleShowImages}
+                onMouseEnter={showImagesF}
+              >
+                <div>{setImageLabel}</div>
+                <div className="icon-codap-inspectorArrow-collapse"/>
+              </li>
+
+              { isTimeBased &&
+                <>
+                  <li
+                    className={collectorClasses}
+                    onClick={this.convertToCollector}
+                    onMouseEnter={hideImagesF}
+                  >
+                    {collectorLabel}
+                  </li>
+
+                  <li
+                    className={flowClasses}
+                    onClick={this.convertToFlow}
+                    onMouseEnter={hideImagesF}
+                  >
+                    {flowLabel}
+                  </li>
+
+                  <li
+                    className={standardClasses}
+                    onClick={this.convertToStandard}
+                    onMouseEnter={hideImagesF}
+                  >
+                  {standardVariableLabel}
+                  </li>
+                </>
+              }
+
+              {showGraphButton &&
+                <li
+                  className={createGraphClasses}
+                  onClick={this.createGraph}
+                  onMouseEnter={hideImagesF}
+                >
+                  {createGraphLabel}
+                </li>
+              }
+
+            </ul>
+          </div>
+          {true && this.renderRightMenu()}
+        </div>
+      </>
+    );
+  }
+
+  private renderRightMenu() {
+    const thisNode = this.props.node;
+    const selected = thisNode.image;
+    const onImageChage = (i: {image: string}) =>  {
+      GraphStore.changeNode({ image: i.image }, thisNode);
+    };
+
+    const palette = PaletteStore.orderedPalette();
+    const imageAddCallback = (i: {image: string}) => {
+      this.setState({ showImages: true, closeTimer: null });
+      GraphStore.changeNode({ image: i.image }, thisNode);
+    };
+    return (
+      <div className="right-panel">
+        <div className="image-choices">
+          <PaletteAddView
+            label={tr("~PALETTE-INSPECTOR.ADD_IMAGE")}
+            callback={imageAddCallback}
+          />
+          { palette.map((pi, i) =>
+            <div className="image-choice-wrapper" key={i}>
```

Comment:
Can you use something else besides the index (`i`) for the key in case the contents of `PaletteStore.orderedPalette` becomes dynamic in the future?  Maybe something unique in the `pi` object?

Reply:
It looks like the `pi` object has a `uuid` that could be used.